File: problems/python/13. Roman to Integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        rmap = {'I': 1, 'V': 5, 'X': 10, 'L': 50, 'C': 100, 'D': 500, 'M': 1000}
        res = 0
        prev = 0
        
        for c in reversed(s):
            val = rmap[c]
            if val < prev:
                res -= val
            else:
                res += val
            prev = val
            
        return res

File: problems/Rust/13. Roman to Integer.rs
use std::collections::HashMap;

impl Solution {
    pub fn roman_to_int(s: String) -> i32 {
        let rmap: HashMap<char, i32> = [
            ('I', 1), 
            ('V', 5), 
            ('X', 10), 
            ('L', 50),
            ('C', 100), 
            ('D', 500), 
            ('M', 1000),
        ].iter().cloned().collect();
        
        let mut res = 0;
        let mut prev = 0;
        
        for c in s.chars().rev() {
            let val = rmap[&c];
            if val < prev {
                res -= val;
            } else {
                res += val;
            }
            prev = val;
        }
        res
    }
}